src/kryptos/k4/straddling_checkerboard.py

```
@@ -0,0 +1,249 @@
+                        "decoded_length": len(decoded),
+                        "keyword_hits": hits,
+                    })
+
+    # Sort by keyword_hits descending for decoding tests
+    decoding_results = [r for r in results if r.get("test") == "k4_decoding"]
+    decoding_results.sort(key=lambda r: (-r.get("keyword_hits", 0), -r.get("decoded_length", 0)))
+
+    encoding_results = [r for r in results if r.get("test") == "k4_encoding_check"]
+
+    summary: dict[str, Any] = {
+        "status": "null_result" if not any(r.get("keyword_hits", 0) >= 4 for r in decoding_results) else "eureka",
```

Pylint (4.0.6)

[convention] Line too long (114/88)

(C0301)